lang/cs/Org.Apache.REEF.Common/metrics/MutableMetricsLayer/IMutableMetric.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using Org.Apache.REEF.Common.metrics.Api;
+
+namespace Org.Apache.REEF.Common.Metrics.MutableMetricsLayer
+{
+    internal interface IMutableMetric : IObserver<SnapshotRequest>
```

Comment:
Thanks. You *did* get the question, after all. You answered it :)

So, whenever `OnNext(SnapshotRequest)` gets called, a `IMutableMetric` is supposed to send its current value? What do we gain by that compared to a method `T GetCurrentSnapshot()`, which is called by the class that holds references to `IMutableMetric` instances?